### DISCUSSION TITLE
BUG: same person more than once in a single photo

### DISCUSSION THREAD

**Discussion by rayzorben** [immich-app/immich#12922] [Q&A]
### The bug

I have twins, they are not identical. In fact in a lot of photos they look similar but very obvious differences. However, with the default 0.5 distance setting, they are grouped together.

I am trying to balance this to get it more accurately, but in the meantime, I find it a bug that in the same photo, both daughters are labeled as the same person. I think that two faces in the same photo should always get split into separate people. If it needs to identify one that matches a known face, it should use the face with the higher score, and then try to match the other face to the next highest score, and no match if it the distance is too far.

### The OS that Immich Server is running on

Proxmox Portainer

### Version of Immich Server

v1.115.0

### Version of Immich Mobile App

No Mobile

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
NA
```

### Your .env content

```Shell
NA
```

### Reproduction steps

1. Have non identical twins in same photo
2. Use immich
3. Matches the same person more than once in a single photo
...

**Comment by mmomjian** [comment]
Thanks for the report. There are real situations (looking in a mirror, for example) where the same face can appear twice in a photo, so this is not a bug. 

Also #6598

**Reply by mlinko6** [reply]
Also the Synology Photos recognizes the difference and I think that the Immich ML is far better than that from Synology... So sorry if I don't agree with the answer that the faces are identical and the software can't recognise it...

![image](https://github.com/user-attachments/assets/d5e41b2f-2838-4b0b-b76c-4fea2c609ea3)

A clear example of what I'm talking about... The print screen is from Synology Photos.

**Reply by mlinko6** [reply]
I'm ok with the option, that at least I can "flag" one of the faces as "wrong" or assign a new name to it so the software would maybe check that these aren't the same faces especially on the same photo.

**Reply by AnthraxLeprosyPi** [reply]
> Thanks for the report. There are real situations (looking in a mirror, for example) where the same face can appear twice in a photo, so this is not a bug.
> 
> Also #6598

Hey Everybody, 

I have the exact same issue with my children... 
They are 4 years apart and boy/girl - I dialed in the settings enough to have a pretty good separation of the two. But in still quite a lot of images one of them gets tagged twice over the other. 

I thought about the mirror argument (in fact my wife brought it up as well) but to me it just doesn't make sense to default to tagging the "same" person twice in an image - even though it might be a mirror image (which the AI should recognize as mirrored ;-))... So I'd love to be able to sort these corner cases out and handle them manually if no other logic applies.

I'd argue it's all about finding the photo when looking for a specific person.

just my two cents :-)

Anthrax

**Comment by rayzorben** [comment]
I don't disagree with what you are saying, but I think it could be tweaked a bit to support both cases.

Face 1 find nearest match
Face 2 find next nearest match within a distance. If no next nearest match within a specific distance, tag as Face 1.

I don't know quite how DBSCAN/scores work here, but here is what I am proposing

Two different people:  
  Face 1 highest score matches to Daughter1 .89
  Face 2 highest score matches to Daughter1 .87 but already exists in photo.
  Face 2 next highest score matches to Daughter2 .83 and is within .1 of Daughter1, so it is tagged as Daughter2

Two of same person in mirror:
  Face 1 highest score matches to Person1 .78
  Face 2 highest score matches to Person1 .77 but already exists in photo.
  Face 2 next highest score matches to Person3 .52 but is too far, so it is tagged as Person1.

**Reply by bo0tzz** [reply]
@mertalev do you think something like this is feasible?

**Reply by mertalev** [reply]
This is feasible. A possible issue for the mirror handling though is if the range would include a different person. I'm not sure how common this is (non-identical siblings?).

**Reply by rayzorben** [reply]
Well a good majority of siblings are non identical :)

About 23 in every 1000 births are fraternal, a much higher rate than the three or four out of 1000 births for identical twins.

So 996 out of 1000 births would not be identical twins.

**Reply by mlinko6** [reply]
> I don't disagree with what you are saying, but I think it could be tweaked a bit to support both cases.
> 
> Face 1 find nearest match Face 2 find next nearest match within a distance. If no next nearest match within a specific distance, tag as Face 1.
> 
> I don't know quite how DBSCAN/scores work here, but here is what I am proposing
> 
> Two different people: Face 1 highest score matches to Daughter1 .89 Face 2 highest score matches to Daughter1 .87 but already exists in photo. Face 2 next highest score matches to Daughter2 .83 and is within .1 of Daughter1, so it is tagged as Daughter2
> 
> Two of same person in mirror: Face 1 highest score matches to Person1 .78 Face 2 highest score matches to Person1 .77 but already exists in photo. Face 2 next highest score matches to Person3 .52 but is too far, so it is tagged as Person1.

How can I "untag" a double face recognition on a single photo to "show" the ML Module that this are 2 different people?

**Comment by mlinko6** [comment]
Also the Synology Photos recognizes the difference and I think that the Immich ML is far better than that from Synology... So sorry if I don't agree with the answer that the faces are identical and the software can't recognise it...

**Comment by urbanrider** [comment]
I have the same issue, generally in blurry old group photos where its tagging the same person 3 or 4 times in a photo!
